ext/json/json_encoder.c

```diff
@@ -104,7 +104,11 @@ static inline void php_json_encode_double(smart_str *buf, double d, int options)
 {
 	size_t len;
 	char num[PHP_JSON_DOUBLE_MAX_LENGTH];
-	php_gcvt(d, (int)EG(precision), '.', 'e', &num[0]);
+	if (JSON_G(precision) == -1) {
+		php_0cvt(d, 17, '.', 'e', num);
+	} else {
+		php_gcvt(d, JSON_G(precision), '.', 'e', num);
+	}
```

Comment:
yeah definitely agree with that. it would be much better to handle that in php_gcvt. there is no need for precision (17 in php_0cvt) for mode 0 because it's not used as you can see here: https://github.com/php/php-src/blob/f6d196a38f484603452ca0ffaa241940c452ac99/Zend/zend_strtod.c#L3940

so you can do everything in php_gcvt (checking -1)...